supervisord/ci/supervisord.rake

```diff
@@ -5,42 +5,40 @@ def supervisord_version
 end
 
 def supervisord_rootdir
-  "#{ENV['INTEGRATIONS_DIR']}/supervisord_#{supervisord_version}"
+  if ENV['INTEGRATIONS_DIR']
+    integrations_dir = ENV['INTEGRATIONS_DIR']
+  else
+    integrations_dir = 'embedded'
+  end
+  "#{integrations_dir}/supervisord"
 end
 
+container_name='dd-test-supervisord'
+container_port=19001
+
 namespace :ci do
   namespace :supervisord do |flavor|
     task before_install: ['ci:common:before_install'] do
-      unless Dir.exist? File.expand_path(supervisord_rootdir)
-        sh %(pip install supervisor==#{supervisord_version} --ignore-installed\
-             --install-option="--prefix=#{supervisord_rootdir}")
-      end
+      sh %(rm -rf #{supervisord_rootdir} || true)
+      sh %(mkdir -p #{supervisord_rootdir} || true)
+      sh %(docker kill #{container_name} 2>/dev/null || true)
+      sh %(docker rm #{container_name} 2>/dev/null || true)
     end
 
     task install: ['ci:common:install'] do
       use_venv = in_venv
       install_requirements('supervisord/requirements.txt',
                            "--cache-dir #{ENV['PIP_CACHE']}",
                            "#{ENV['VOLATILE_DIR']}/ci.log", use_venv)
+      sh %(docker run -d --name #{container_name} -p #{container_port}:#{container_port} -v #{supervisord_rootdir}:/supervisor datadog/docker-library:supervisord_3_3_0)
     end
 
     task before_script: ['ci:common:before_script'] do
-      sh %(mkdir -p $VOLATILE_DIR/supervisor)
-      sh %(cp supervisord/ci/resources/supervisord.conf $VOLATILE_DIR/supervisor/)
-      sh %(sed -i -- 's/VOLATILE_DIR/#{ENV['VOLATILE_DIR'].gsub '/', '\/'}/g'\
-         $VOLATILE_DIR/supervisor/supervisord.conf)
-
-      3.times do |i|
-        sh %(cp supervisord/ci/resources/program_#{i}.sh $VOLATILE_DIR/supervisor/)
-      end
-      sh %(chmod a+x $VOLATILE_DIR/supervisor/program_*.sh)
-
-      sh %(#{supervisord_rootdir}/bin/supervisord\
-           -c $VOLATILE_DIR/supervisor/supervisord.conf)
-      3.times { |i| Wait.for "#{ENV['VOLATILE_DIR']}/supervisor/started_#{i}" }
+      # we need to make sure that supervisor is running an the rpc port is up
+      Wait.for 19001
       # And we still have to sleep a little, because sometimes supervisor
       # doesn't immediately realize that its processes are running
-      sleep_for 1
+      sleep_for 2
```

Comment:
should we do this? Or is there anything in the logs we can wait for (just to make it more resilient)?